protocols/imap/src/main/java/org/apache/james/imap/decode/ImapRequestLineReader.java

```diff
@@ -814,6 +815,24 @@ private long parseUnsignedInteger(String value) throws DecodingException {
         boolean isValid(char chr);
     }
 
+    public static class StringValidator implements ImapRequestLineReader.CharacterValidator {
+        private final byte[] expectedStringAsBytes;
+        private int position = 0;
+
+        public StringValidator(String expectedString) {
+            this.expectedStringAsBytes = expectedString.getBytes(StandardCharsets.US_ASCII);
```

Comment:
why do you want to convert the String as byte[] when you want to compare characters?
BTW, `ImapRequestLineReader.cap(chr)` is insane, this whole class don't really make sense because of that.
Please write some unit tests about this class.